tests/integration/targets/filesize/tasks/sparse.yml

```
@@ -0,0 +1,167 @@
+---
+# Test module with sparse files
+
+- name: Create a huge sparse file of 4TB (check mode)
+  filesize:
+    path: "{{ filesize_testfile }}"
+    size: "4TB"
+    sparse: yes
+  register: filesize_test_sparse_01
+  check_mode: yes
+
+- name: Create a huge sparse file of 4TB
+  filesize:
+    path: "{{ filesize_testfile }}"
+    size: "4TB"
+    sparse: yes
+  register: filesize_test_sparse_02
+
+- name: Create a huge sparse file of 4TB (4000GB) (check mode, idempotency)
+  filesize:
+    path: "{{ filesize_testfile }}"
+    size: "4000GB"
+    sparse: yes
+  register: filesize_test_sparse_03
+  check_mode: yes
+
+- name: Create a huge sparse file of 4TB (4000GB) (idempotency)
+  filesize:
+    path: "{{ filesize_testfile }}"
+    size: "4000GB"
+    sparse: yes
+  register: filesize_test_sparse_04
+
+- name: Create a huge sparse file of 4TB (4000000 × 1MB) (check mode, idempotency)
+  filesize:
+    path: "{{ filesize_testfile }}"
+    size: "4000000"
+    blocksize: 1MB
+    sparse: yes
+  register: filesize_test_sparse_05
+  check_mode: yes
+
+- name: Create a huge sparse file of 4TB (4000000 × 1MB) (idempotency)
+  filesize:
+    path: "{{ filesize_testfile }}"
+    size: "4000000"
+    blocksize: 1MB
+    sparse: yes
+  register: filesize_test_sparse_06
+
+- name: Assert that results are as expected
```

```suggestion
- name: Assert that results are as expected
```
shouldn't we check the actual state (the files exist and have expected size) with something else? For example, with the `stat` module ?

---

yes, sounds better. I wiil do

---

thanks!